src/main/java/org/primeframework/mvc/parameter/annotation/NamedParameter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023, Inversoft Inc., All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package org.primeframework.mvc.parameter.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation allows you to specify the name of a parameter to be parsed from the request that is different from field name on the target class.
+ *
+ * @author Lyle Schemmerling
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface NamedParameter {
+  String name();
```

Comment:
One possible thought here would be to use `value` instead of name so that you don't have to specify the name when using it. 

For example, now usage looks like:

```
@Named(name = "foo")
```

If you use `value` instead it would look like:

```
@Named("foo")
```

For example, we are essentially doing the same thing that `@JsonProperty` does in Jackson. Here is that annotation:

```
public @interface JsonProperty {
  String value() default "";
}
```

This allows you to use it like `@JsonProperty("foo")`